src/adagrams.js

```diff
@@ -1,7 +1,148 @@
 const Adagrams = {
   drawLetters() {
-    // Implement this method for wave 1
+    const lettersDrawn = []
+    const letterDistribution = {
+      "A": 9,
+      "B": 2,
+      "C": 2,
+      "D": 4,
+      "E": 12,
+      "F": 2,
+      "G": 3,
+      "H": 2,
+      "I": 9,
+      "J": 1,
+      "K": 1,
+      "L": 4,
+      "M": 2,
+      "N": 6,
+      "O": 8,
+      "P": 2,
+      "Q": 1,
+      "R": 6,
+      "S": 4,
+      "T": 6,
+      "U": 4,
+      "V": 2,
+      "W": 2,
+      "X": 1,
+      "Y": 2,
+      "Z": 1,
+      pickRandomLetter(){
+        return Object.keys(this)[Math.floor(Math.random() * Math.floor(Object.keys(this).length - 1))];
+      },
+    }
+
+    for (let i = 0; i < 10; i++) {
+      let letter = letterDistribution.pickRandomLetter();
+
+      while(letterDistribution[letter] === 0) {
+        letter = letterDistribution.pickRandomLetter();
+      }
+
+      lettersDrawn.push(letter);
+      letterDistribution[letter] -= 1;
+    }
+   
+    return lettersDrawn;
+  },
```

Comment:
This doesn't actually create a weighted distribution. You are essentially picking from a pile 26 options, and then reducing the number of times that someone can draw that letter, which is mathematically different than having a pile that has a hundred or so tiles and just one 'Q'.

Reply:
Oh no... >_<  I'm not sure I understand what you mean or what I should have done differently to use weighted distribution. I am googling its meaning~!  Are you saying I should rename my variables or change my function???

Reply:
Ok, so should I have made an object with the total amount of tiles for the all the letters and then give each letter a unique number.. like instead of {A:9, B:2} for a total of 11 tiles, should I have done {A:1, A:2, A:3...B:10, B:11}???